src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -190,12 +194,22 @@ private bool EnableInstrumentation
                 layoutPage.RenderBodyDelegate = bodyWriter.CopyTo;
                 bodyWriter = await RenderPageAsync(layoutPage, context, executeViewStart: false);
 
-                // Verify that RenderBody is called, or that RenderSection is called for all sections
-                layoutPage.EnsureBodyAndSectionsWereRendered();
+                // Verify that RenderBody is called
+                layoutPage.EnsureBodyWasRendered();
+
+                unrenderedSections.UnionWith(layoutPage.PreviousSectionWriters.Keys);
```

Comment:
This might not address sections that are redefined but not rendered in the outer definition.

```C#
// NestedLayout
@{ Layout = "OuterLayout"; }
@section scripts {
   // stuff
}

// OuterLayout
@{ Layout = "BaseLayout"; }
@section scripts { 
  // Don't render inner scripts
}

// BaseLayout
@RenderSection("scripts");
```

In this scenario, the `scripts` section defined by `InnerLayout` is never rendered but the view would happily succeed. Perhaps we shouldn't be flattening the results in `unrenderedSections`?